tee/tee.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package tee
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type (
+	Enclave interface {
+		// GenKeys generates a new secp256k1 ECDSA key and stores it as the
+		// enclave's signing key.
+		//
+		// It returns the public key derived Ethereum address and attestation of
+		// correct initialization of the enclave with the generated address. The
+		// attestation can be used to verify the enclave with the TEE vendor.
+		//
+		// The Operator must deploy the contract with the Enclave's address after
+		// calling GenKeys.
+		GenKeys() (common.Address, []byte, error)
```

Comment:
I think that GenKey() or Setup() or Init() or something like that would be better here (as only one key is generated). I would also name the returned values OR make a type alias for the attestation.

Reply:
Having it called `Init()` would also be more in line with other SGX implementations I saw.

Reply:
Guys, this is a hackathon, lower your standards ;) Renamed to `Init`.